lib/gooddata/models/data_source.rb

```diff
@@ -0,0 +1,102 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+#
+# Copyright (c) 2010-2020 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+module GoodData
+  class DataSource < Rest::Resource
+    class << self
```

Comment:
models/data_source.rb should represent for dataSource object with functions to CRUD:
- constructor: create new datasource object or load it from server.
- create: create new datasource.
- save: create or update datasource
- all: list data sources

You can inspire from other resources process.rb or schedule.rb